pwiz_tools/Skyline/Model/Results/ChromHeaderInfo.cs

```diff
@@ -1920,7 +1920,18 @@ public static ChromKey FromId(string idIn, bool parseCE)
                 }
                 else if (id.StartsWith(MsDataFileImpl.PREFIX_PRECURSOR))
                 {
-                    precursor = double.Parse(id.Substring(MsDataFileImpl.PREFIX_TOTAL.Length), CultureInfo.InvariantCulture);
+                    var str = id.Substring(MsDataFileImpl.PREFIX_PRECURSOR.Length);
+                    if (!double.TryParse(str, NumberStyles.Float, CultureInfo.InvariantCulture, out precursor))
+                    {
+                        if (str.StartsWith(@"Q1="))
+                        {
+                            str = str.Substring(3);
+                            var tail = str.IndexOf(' ');

```

Comment:
str.Substring(3).Split(' ')[0] would also work and is more concise.

Reply:
Sounds good.